src/hidpp10.c

```diff
@@ -2109,12 +2109,23 @@ hidpp10_set_current_resolution(struct hidpp10_device *dev,
 			       uint16_t yres)
 {
 	unsigned idx = dev->index;
-	union hidpp10_message resolution = CMD_CURRENT_RESOLUTION(REPORT_ID_LONG, idx, SET_LONG_REGISTER_REQ);
+	union hidpp10_message resolution = CMD_CURRENT_RESOLUTION(REPORT_ID_SHORT, idx, SET_REGISTER_REQ);
+	union hidpp10_message resolution_long = CMD_CURRENT_RESOLUTION(REPORT_ID_LONG, idx, SET_LONG_REGISTER_REQ);
+	int res;
+
+	switch (dev->profile_type) {
+	case HIDPP10_PROFILE_G9:
+		hidpp_set_unaligned_le_u16(&resolution.data[4], hidpp10_get_dpi_mapping(dev, xres));
```

Comment:
Shouldn't it be a 8 bit value?

Reply:
Good catch. It should. hidpp10_get_dpi_mapping returns a uint8_t so we can just write that directly. The result is the same but it is more correct. I'll force push an updated patch